fix(side_bar): use element based tooltips position

dqunbp opened the pull request:
Use element-based positioning for sidebar tooltips 
Extract NavButton component
Extract ToggleButton component

https://kontur.fibery.io/Tasks/Task/Wrong-tooltips-positioning-on-DN-sidebar-buttons-14909

dqunbp committed:
fix(side_bar): use element based tooltips position

Use element based positioning for sidebar tooltips
Extract NavButton component
Extract ToggleButton coponent

Akiyamka commented on src/features/side_bar/components/SideBar/NavButton.tsx:
```diff
@@ -0,0 +1,81 @@
+  const [searchString] = useAtom(searchStringAtom);
+  const [currentRoute] = useAtom(currentRouteAtom);
+
+  const isVisible = checkRouteVisibility(route, currentRoute);
```
Why not pass isVisible as boolean prop? Right now this function invokes on every render

Akiyamka commented on src/features/side_bar/components/SideBar/NavButton.tsx:
```diff
@@ -0,0 +1,81 @@
+
+  if (!isVisible) return null;
+
+  const to = getAbsoluteRoute(
```
Looks like router logic

Akiyamka commented on src/features/side_bar/components/SideBar/NavButton.tsx:
```diff
@@ -0,0 +1,81 @@
+      : `${route.slug}${searchString}`,
+  );
+
+  const className = cn(
```
any element  may have className prop. If possible - write this logic in jsx, in you need to move it in variable - give them some more unique name

dqunbp committed:
refactor: nav-button changes

Update getAbsoluteRoute to handle routes as well
Add isVisible prop to NavButton
